core/src/main/java/org/apache/brooklyn/core/entity/AbstractEntity.java

```diff
@@ -1548,7 +1550,7 @@ public boolean unsubscribe(Entity producer, SubscriptionHandle handle) {
      * @deprecated since 0.9.0; for internal use only
      */
     @Deprecated
-    protected synchronized SubscriptionTracker getSubscriptionTracker() {
+    protected SubscriptionTracker getSubscriptionTracker() {
```

Comment:
the `BasicSubscriptionSupport.getSubscriptionTracker()` method this calls also synchronizes on `AbstractEntity.this` (line 1494 in new money) -- so no impact whether this method is `synchronized` or not

in contrast to the change to unsynchronize `getSubscriptionContext()` here, synchronizing on *something* when getting the tracker looks right as it is setting a local field.  however the underlying method (line 1498 new money) calls to `getSubscriptionContext()` while synchronized on `this` so i think we haven't entirely eliminated the deadlock prospect.

feels like line 1498 could double-check the tracker, and load the `getSubscriptionContext()` between the checks eg

    if (tracker!=null) return tracker;
    subs = getSubscriptionContext();
    synchronized (AbstractEntity.this) {
        if (tracker!=null) return tracker;
        tracker = new Tracker(subs);
        return tracker;
    }